async-book/src/execution/wakeups.md

```diff
@@ -0,0 +1,143 @@
+# Task Wakeups with `LocalWaker` and `Waker`
+
+It's common that futures aren't able to complete the first time they are
+`poll`ed. When this happens, the future needs to ensure that it is polled
+again once it is ready to make more progress. This is done with the
+`LocalWaker` and `Waker` types.
+
+Each time a future is polled, it is polled as part of a "task". Tasks are
+top-level futures that track readiness and can schedule themselves to be
```

Comment:
maybe: "*the* top-level futures *that have been submitted to an executor".

Reply:
oh see I've been using task to refer to the internal concept that holds both a future and a reference to the executor, and can put itself back on the ready queue. Not just the `Future<Output = ()>`. It's the "lightweight thread" mechanism. However, I think that might be mixing things up a bit too much for this early on, so I'll accept your suggestion ;)

Reply:
Done.